### ISSUE TITLE
drive_ls "recursive" flag broken

### ISSUE THREAD

**Issue by lotard** [tidyverse/googledrive#265]
> drive_ls(..., recursive = T)

Only expands the first "folder" in the path.

**Comment by jennybc** [comment]
Can you provide a more detailed reprex of what you have (in terms of file/directory state on Drive), what call you make, results you expect vs. results you get?

**Comment by lotard** [comment]
Mhm, doesn't seem to be a problem anymore, not sure why, maybe some interference from other packages or environmental variables... Please close this issue, sorry for wasting your time!

**Comment by jennybc** [comment]
Ok great. FYI, you have the power to close it, if you opened it.

**Comment by Falnesio** [comment]
I have had the same problem. @lotard what did you find that caused interference, if you would be so kind?

**Comment by Falnesio** [comment]
@jennybc I fixed it in Falnesio:Fix-recursiveness-in-drive_ls. It wasn't working in shared drives.

**Comment by jennybc** [comment]
@Falnesio Can you add `"Fixes #265"` to your PR to link it to this?

When I come back through to review it, it would still be extremely helpful to have a clean reprex that helps me:

* Experience the problem first hand. Meaning: sets up the situation where we can see `drive_ls()` fail. I can imagine what's going wrong, but it's so much faster if someone who is suffering has already constructed the perfect example.
* Experience the joy of seeing it fixed by the PR first hand.

**Comment by Falnesio** [comment]
This folder has been made public.
https://drive.google.com/drive/u/0/folders/1QPDu46aGHh8EjdJm-i12NzcVVBlVjyvt

![image](https://user-images.githubusercontent.com/36005374/157115174-651ae148-55b1-4908-be16-8853f6001118.png)


**Before fix:**
```
> drive_ls(as_id("1QPDu46aGHh8EjdJm-i12NzcVVBlVjyvt"), recursive = TRUE, type="folder")
# A dribble: 3 x 3
  name  id                                drive_resource   
  <chr> <drv_id>                          <list>           
1 C     1iuIBsZTQNQPfdYkFqRdsbHX8fcpjb3ya <named list [32]>
2 A     17qBnOnqjP3nemZdo5bcY4htnp4IZJrxg <named list [32]>
3 B     186JZdflymqlq-DVa8S0jXbMGhcTEF5Vv <named list [32]>

```
**After fix:**
```
> drive_ls(as_id("1QPDu46aGHh8EjdJm-i12NzcVVBlVjyvt"), recursive = TRUE, type="folder")
# A dribble: 6 x 3
  name  id                                drive_resource   
  <chr> <drv_id>                          <list>           
1 B.3   1N9BRc3DNfBNa3csrbe8vnmBTO0f8IOSa <named list [32]>
2 C     1iuIBsZTQNQPfdYkFqRdsbHX8fcpjb3ya <named list [32]>
3 B.2   1Q0o-D6r8xJgIOkyqkrGCLk5vGOdR-Rkg <named list [32]>
4 A     17qBnOnqjP3nemZdo5bcY4htnp4IZJrxg <named list [32]>
5 A.1   1Qy6-CS5KAbYTuvu4jZjj7rL90j9kdg1O <named list [32]>
6 B     186JZdflymqlq-DVa8S0jXbMGhcTEF5Vv <named list [32]>
```
